### PR TITLE
Add dummy image dataset generator helper class

### PR DESCRIPTION
**Pull request by samet-akcay**
## Description

This PR adds `DummyImageGenerator` and `DummyDatasetGenerator` classes to test helpers to generate the datasets for the tests.

## Changes

<details>
<summary>Describe the changes you made</summary>

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

File: src/anomalib/data/__init__.py [hunk 1]
```diff
@@ -28,20 +29,32 @@
...
+class VideoDataFormat(str, Enum):
+    """Supported Video Dataset Types."""
+
     UCSDPED = "ucsdped"
     AVENUE = "avenue"
-    VISA = "visa"
     SHANGHAITECH = "shanghaitech"
 
 
+DataFormat = Enum(  # type: ignore[misc]
```
Comment (ashwinvaidya17) [inline src/anomalib/data/__init__.py:52]: Can the type system parse this?

File: tests/helpers/dataset.py [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+        >>> image, mask = generator.generate_image(label=LabelName.NORMAL)
+    """
+
+    def __init__(self, image_shape: tuple[int, int] = (256, 256)) -> None:
+        self.image_shape = image_shape
+        self.augmenter = Augmenter()
+
+    def generate_normal_image(self) -> tuple[np.ndarray, np.ndarray]:
+        """Generate a normal image."""
+        image = random_shapes(image_shape=self.image_shape, min_size=256, max_shapes=1, shape="rectangle")[0]
```
Comment (ashwinvaidya17) [inline tests/helpers/dataset.py:54]: If it's just a full rectangle do we need `random_shapes`? Maybe we can do something like
```python
image = = np.zeros([*self.image_shape, 3])
image[...] = np.random.randint(low=0, high=255, size=[3])
```

**Review by ashwinvaidya17 — COMMENTED** [review]
I haven't run it yet but I have two very minor comments. But the code looks really clean. I am looking forward to replacing the previous mess with this.

File: src/anomalib/data/__init__.py (hunk 1 above)
Comment (samet-akcay) [reply src/anomalib/data/__init__.py:52]: Mypy does not complain, but Pylance complains about the following:
```
Variable not allowed in type expression
```

**Pull request merged**